crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -843,6 +865,112 @@ impl AuthorityPerEpochStore {
             Either::Right((result, _)) => Ok(result),
         }
     }
+
+    pub fn get_pending_checkpoints(
+        &self,
+    ) -> Vec<(CheckpointCommitHeight, (Vec<TransactionDigest>, bool))> {
+        self.tables.pending_checkpoints.iter().collect()
+    }
+
+    pub fn get_pending_checkpoint(
+        &self,
+        index: &CheckpointCommitHeight,
+    ) -> Result<Option<(Vec<TransactionDigest>, bool)>, TypedStoreError> {
+        self.tables.pending_checkpoints.get(index)
+    }
+
+    pub fn insert_pending_checkpoint(
+        &self,
+        index: &CheckpointCommitHeight,
+        transactions: &(Vec<TransactionDigest>, bool),
+    ) -> Result<(), TypedStoreError> {
+        self.tables.pending_checkpoints.insert(index, transactions)
+    }
+
+    pub fn process_pending_checkpoint(
+        &self,
+        commit_height: CheckpointCommitHeight,
+        content_info: Option<(CheckpointSequenceNumber, Vec<TransactionDigest>)>,
+    ) -> Result<(), TypedStoreError> {
+        let mut batch = self.tables.pending_checkpoints.batch();
+        batch = batch.delete_batch(&self.tables.pending_checkpoints, [commit_height])?;
+        if let Some((seq, transactions)) = content_info {
+            batch = batch.insert_batch(
+                &self.tables.digest_to_checkpoint,
+                transactions.iter().map(|tx| (*tx, seq)),
+            )?;
+        }
+
+        batch.write()
+    }
+
+    pub fn tx_checkpointed_in_current_epoch(
+        &self,
+        digest: &TransactionDigest,
+    ) -> Result<bool, TypedStoreError> {
+        self.tables.digest_to_checkpoint.contains_key(digest)
+    }
+
+    pub fn aggregate_pending_checkpoint_signatures(
```

Comment:
I think it would be better if `AuthorityEpochTables` just expose a function like 
`pending_checkpoint_singatures(checkpoint_sequence: u64, from_signature_index_included: u64) -> impl Iter<(u64/*index*/, CheckpointSignatureMessage)>`

I think this way the `CheckpointSignatureAggregator` can stay within checkpoint code, and it will be in line with general code pattern where epoch_store do not do much except for reading db.

Reply:
Done